src/js/web-components/video-player/VideoPlayer.js

```diff
@@ -139,19 +139,24 @@ export default class extends HTMLElement {
 
             const request = new window.chrome.cast.media.LoadRequest(mediaInfo);
 
-            castSession.loadMedia(request).then(
-              () => {
-                const targetName = castSession.getCastDevice().friendlyName;
-
-                if (targetName) {
-                  this.internal.root.querySelector(`.${CAST_TARGET_NAME}`).innerText = targetName;
-                  this.classList.add(CAST_HAS_TARGET_NAME);
-                }
-                this.classList.add(CAST_CLASSNAME);
-              },
+            try {
+              await castSession.loadMedia(request);
+            } catch (error) {
               /* eslint-disable-next-line no-console */
-              (errorCode) => console.error(`[Google Cast] Error code: ${errorCode}`),
-            );
+              console.error(`[Google Cast] Error code: ${error}`);
+              return;
+            }
+
+            const targetName = castSession.getCastDevice().friendlyName;
+
+            if (targetName) {
+              this.internal.root.querySelector(`.${CAST_TARGET_NAME}`).innerText = targetName;
```

Comment:
One last nit. What do you think of this?

```js
            const targetName = castSession.getCastDevice().friendlyName;
            this.internal.root.querySelector(`.${CAST_TARGET_NAME}`).innerText = targetName;
            this.classList.toggle(CAST_HAS_TARGET_NAME, targetName);
            this.classList.add(CAST_CLASSNAME);
```

```

Reply:
@beaufortfrancois Elegant, thanks!